frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -392,16 +393,28 @@ class MidCircuitMeasure(HybridOp):
 
     binder = qmeasure_p.bind
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.bypass_postselect = kwargs.pop("bypass_postselect", False)
+        self.postselect = self.in_classical_tracers[-2]
+        self.reset = self.in_classical_tracers[-1]
+
     def trace_quantum(self, ctx, device, trace, qrp) -> QRegPromise:
         op = self
         wire = op.in_classical_tracers[0]
         qubit = qrp.extract([wire])[0]
-        postselect = op.in_classical_tracers[1]
-
-        qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=postselect)
+        if self.bypass_postselect:
+            qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit)
+        else:
+            postselect = op.in_classical_tracers[1]
+            qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=postselect)
         qrp.insert([wire], [qubit2])
         return qrp
 
+    def __hash__(self):
+        hsh = super().__hash__()
+        return hash(hsh + hash(self.out_classical_tracers[0]))
```

Comment:
Just curious how you decided on this hash implementation :)

Reply:
The original hash function didn't distinguish between MCMs. The first `out_classical_tracers` is the MCM value, which is enough to identify MCMs for the purpose of post-processing. So I thought I would just hash the default hash value together with that of the MCM value.

Reply:
Good point, that makes sense I suppose 👍